internal/resource/http.go

```diff
@@ -272,3 +281,31 @@ func (c HttpClient) getReaderWithHeader(url string, header http.Header) (io.Read
 		}
 	}
 }
+
+func proxyFuncFromIgnitionConfig(proxy types.Proxy) func(*url.URL) (*url.URL, error) {
+	noProxy := translateNoProxySliceToString(proxy.NoProxy)
+
+	if proxy.HTTPProxy == nil {
+		proxy.HTTPProxy = new(string)
+	}
+
+	if proxy.HTTPSProxy == nil {
+		proxy.HTTPSProxy = new(string)
+	}
+
+	cfg := &httpproxy.Config{
```

Comment:
does this need to be a `&`?

Reply:
Hmm, not sure, I'll also research when investigating the test failure.

Reply:
I remember why I used `&` now. I was using the same convention as [FromEnvironment](https://go.googlesource.com/net/+/refs/heads/release-branch.go1.12/http/httpproxy/proxy.go#91).